server/service/live_queries.go

```diff
@@ -55,14 +56,34 @@ func runLiveQueryEndpoint(ctx context.Context, request interface{}, svc fleet.Se
 		},
 	}
 
-	queryResults, respondedHostCount := svc.RunLiveQueryDeadline(ctx, req.QueryIDs, req.HostIDs, duration)
+	queryResults, respondedHostCount, err := svc.RunLiveQueryDeadline(ctx, req.QueryIDs, req.HostIDs, duration)
+	if err != nil {
+		return nil, err
+	}
+	// Check if all query results were forbidden due to lack of authorization
+	allResultsForbidden := len(queryResults) > 0
+	for _, r := range queryResults {
+		if r.Error == nil || *r.Error != authz.ForbiddenErrorMessage {
+			allResultsForbidden = false
+			break
+		}
+	}
+	if allResultsForbidden {
+		return nil, authz.ForbiddenWithInternal("All Live Query results were forbidden.", authz.UserFromContext(ctx), nil, nil)
+	}
```

Comment:
Let's check with the product team what the expected behavior is when sending multiple `query_ids` and the user is not authorized to run some of them:
- Should the request fail and no queries be executed?
- Should the request not fail and only run the queries that the user is authorized to run?

API: https://fleetdm.com/docs/rest-api/rest-api#parameters97

@noahtalerman @rachaelshaw

Reply:
@marko-lisica did we define a similar behavior for running MDM commands that we can borrow here?

Reply:
Current behavior for mix of authorized/unauthorized live queries is that user will get back an array of results. Good results will be valid, and unauthorized results will have `"error":"forbidden"`

Reply:
Sounds good. Could we take the chance to document this behavior in the rest-api.md?

Reply:
(Can be done later on another PR.)

Reply:
This behavior of mixed results makes sense to me but I definitely agree we should document the behavior. @getvictor if you don't mind adding that to this PR, that'd be awesome. Or I'd be happy to take a stab at it after this is merged, just let me know

Reply:
@noahtalerman In the CLI we have error message for this use case - [figma link](https://www.figma.com/file/6eRuzbDWIUCLPdixIqdmea/%2313069-Windows-MDM-protocol-commands?type=design&node-id=142-238&mode=design&t=9oiJf2ohl6LtfaFX-11). Regarding API, seems there's [403: forbidden](https://github.com/fleetdm/fleet/issues/13595#:~:text=403%3A%20Error%3A%20Forbidden) error, but not sure when do we return this one.

Reply:
@rachaelshaw I added PR https://github.com/fleetdm/fleet/pull/14956 for rest-api.md updates.